src/Symfony/Component/Mailer/Transport/Smtp/Auth/AuthenticatorInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Transport\Smtp\Auth;
+
+use Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport;
+
+/**
+ * An Authentication mechanism.
+ *
+ * @author Chris Corbyn
+ *
+ * @experimental in 4.3
+ */
+interface AuthenticatorInterface
+{
+    /**
+     * Tries to authenticate the user with $username and $password.
+     */
+    public function authenticate(EsmtpTransport $client);
```

Comment:
```suggestion
    public function authenticate(EsmtpTransport $client) void;
```

Or do we expect a return value here? If yes, it should be documented in the doc block at least.

Reply:
no return value expected.